crates/telio-starcast/src/transport.rs

```diff
@@ -228,6 +281,29 @@ impl Runtime for State {
     type Err = Error;
 
     async fn wait(&mut self) -> WaitResponse<'_, Self::Err> {
+        let transport_socket = match &self.transport_socket {
+            Some(transport_socket) => transport_socket.to_owned(),
+            None => {
+                PinnedSleep::new(self.exponential_backoff.get_backoff(), ()).await;
+                let Ok(transport_socket) = open_transport_port(
+                    self.socket_pool.clone(),
+                    self.meshnet_ip,
+                    MULTICAST_TRANSPORT_PORT,
+                )
```

Comment:
Do you assign it somehow to `self.transport_socket`?

Reply:
Lost that part during refactoring. Fixed now.

Reply:
Well, maybe you could add some test for this, so it won't be lost during the next refactor? ;)

Reply:
Added a test for it